src/main/java/com/google/api/codegen/transformer/csharp/CSharpModelTypeNameConverter.java

```diff
@@ -107,7 +132,29 @@ public TypeName getTypeName(ProtoElement elem) {
 
   @Override
   public TypedValue getZeroValue(TypeRef type) {
-    return TypedValue.create(getTypeName(type), "ZERO_VALUE"); // TODO: Generate correctly
+    if (type.isMap()) {
+      TypeName keyTypeName = getTypeName(type.getMapKeyField());
+      TypeName valueTypeName = getTypeName(type.getMapValueField());
+      String pattern =
+          "new %s<" + keyTypeName.getNickname() + ", " + valueTypeName.getNickname() + ">()";
+      return TypedValue.create(
+          typeNameConverter.getTypeName("System.Collections.Generic.Dictionary"), pattern);
```

Comment:
This implementation loses the ability to defer name clash resolution because it gets the nicknames without consulting the type table. I think you need two layers of templating here (and below too). You need almost what is in getTypeName(TypeRef) to get the TypeName to pass to TypedValue, except you would want Dictionary instead of IDictionary as the type name. So something like:

TypeName implTypeName = getImplTypeName(type); // uses Dictionary instead of IDictionary
return TypedValue.create(implTypeName, "new %s()";